### PR TITLE
[Ruby][faraday] fix response streaming

### PR DESCRIPTION
**Pull request by wing328**
To fix https://github.com/OpenAPITools/openapi-generator/issues/7708

Tested locally to confirm the fix.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

cc @cliffano (2017/07) @zlx (2017/09) @autopp (2019/02)

### CODE REVIEW

File: samples/client/petstore/ruby-faraday/lib/petstore/api_client.rb [hunk 1]
```diff
@@ -200,19 +200,22 @@ def download_file(request)
         @tempfile = tempfile
       end
 
+      @stream = []
+
       # handle streaming Responses
       request.options.on_data = Proc.new do |chunk, overall_received_bytes|
         # puts "Received #{overall_received_bytes} characters"
-        tempfile.write(chunk)
+        @stream << chunk
```
Comment (tsuwatch) [inline samples/client/petstore/ruby-faraday/lib/petstore/api_client.rb:207]: If you use a block, the file will be closed automatically.
Can this be used?
https://docs.ruby-lang.org/en/2.0.0/Net/HTTP.html#class-Net::HTTP-label-Streaming+Response+Bodies

Comment (wing328) [reply samples/client/petstore/ruby-faraday/lib/petstore/api_client.rb:207]: Could be. Previously I got IO error when using `tempfile.write(chunk)`.

Can you test your suggested fix locally and if it works, can you please file a PR?

Comment (tsuwatch) [reply samples/client/petstore/ruby-faraday/lib/petstore/api_client.rb:207]: I see, I got it.

**Comment by tsuwatch** [comment]
Thank you very much for fixing my issue.

**Pull request merged**